### PR TITLE
NETOBSERV-872 NetObserv Operator installation and e2e-test

### PR DESCRIPTION
**Pull request by memodi**
[NETOBSERV-872](https://issues.redhat.com/browse/NETOBSERV-872) NetObserv Operator installation such that it can be deployed with cluster bot and also adding e2e-test 

Some commented changes in e2e-test depends on PR https://github.com/netobserv/network-observability-operator/pull/535/

### CODE REVIEW

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
I will uncomment some lines in test steps once PR #https://github.com/netobserv/network-observability-operator/pull/535 is merged.

But this should be ready to review. @openshift/netobserv-team

**Comment by memodi** [comment]
/hold 
(for https://github.com/netobserv/network-observability-operator/pull/535)

File: ci-operator/config/netobserv/network-observability-operator/netobserv-network-observability-operator-main.yaml [hunk 1]
```diff
@@ -45,14 +45,44 @@ tests:
...
+        USER=netobserv VERSION=main make set-flp-image
+        # USER=netobserv VERSION=main make set-plugin-image
+        # make set-release-kind-downstream
+        PORT_FWD=false NAMESPACE=netobserv make deploy-loki
+        oc apply -f /go/src/github.com/netobserv/network-observability-operator/config/samples/flows_v1beta2_flowcollector.yaml
+        sleep 5
+
+        oc wait pods --timeout=120s --for=condition=ready -l app=flowlogs-pipeline -n netobserv
+        # oc wait -n netobserv --timeout=60s --for condition=Available=True deployment netobserv-plugin
+        oc wait pods --timeout=120s --for=condition=ready -l app=netobserv-ebpf-agent -n netobserv-privileged
```
Comment (jpinsonneau) [inline ci-operator/config/netobserv/network-observability-operator/netobserv-network-observability-operator-main.yaml:76]: Something you could read here is simply the `Status` of the `FlowCollector` CR

Comment (memodi) [reply ci-operator/config/netobserv/network-observability-operator/netobserv-network-observability-operator-main.yaml]: good idea, I will add that. thanks!

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
/pj-rehearse pull-ci-netobserv-network-observability-operator-main-e2e-operator

**Comment by memodi** [comment]
@jpinsonneau @jotak - the e2e-operator rehearsal passed, could you PTAL?

File: ci-operator/config/netobserv/network-observability-operator/netobserv-network-observability-operator-main.yaml [hunk 2]
```diff
@@ -45,14 +45,44 @@ tests:
...
+        export PATH=$PATH:$HOME
+
+        if ! which kubectl; then
+          ln -s "$(which oc)" ${HOME}/kubectl
+        fi
+
+        export CSV=$(oc get csv -n openshift-netobserv-operator | grep -E "net.*observ" | awk '{print $1}')
+        USER=netobserv VERSION=main make set-agent-image
+        USER=netobserv VERSION=main make set-flp-image
+        USER=netobserv VERSION=main make set-plugin-image
```
Comment (jpinsonneau) [inline ci-operator/config/netobserv/network-observability-operator/netobserv-network-observability-operator-main.yaml:68]: I suppose that's a preparatory work for future capabilities to pass custom image arguments ?

**Review by jpinsonneau — APPROVED** [review]
Looks good to me ! Thanks @memodi

File: ci-operator/config/netobserv/network-observability-operator/netobserv-network-observability-operator-main.yaml (hunk 2 above)
Comment (memodi) [reply ci-operator/config/netobserv/network-observability-operator/netobserv-network-observability-operator-main.yaml:68]: Sorry, I missed this.
Yes, that's right. Once we add configs for other operands, we'd have to find a way to such that it deploys custom images that's build during the test.

**Pull request merged**